### DISCUSSION TITLE
formState.isDirty does not updating after reset/setValue.

### DISCUSSION THREAD

**Discussion by KhorenGrig** [react-hook-form/react-hook-form#4461] [Q&A]
I am initializing the form with default values, and I have two buttons. First one `resets` the form and the second one `fills` the form. The question is before these actions I am logging `formState.isDirty` and it's logging boolean value corresponding `form is dirty` or not, but after `reset/set`, it does not log any value. What can be a reason? 

Sandbox link https://codesandbox.io/s/react-hook-form-v6-setvalue-forked-nujs2?file=/src/index.jsx

**Comment by bluebill1049** [comment]
what do you mean not log any value?

**Reply by KhorenGrig** [reply]
`bluebill1049` Thanks for the reply. If you look at my example, you will see that I am logging `formState.isDiry` prop in `useEffect ` hook. Before any action(reset/setVaue), if you change input value, `log` is triggered, but after this action (reset/setValue), if you  change input value, `log` doesn't trigger.

**Reply by bluebill1049** [reply]
are you putting formState as dep?

```
const { formState, formState:  {isDirty} } = useForm(); // subscribe to isDirty

useEffect(() => {}, [formState]) // formState as dep
```

**Reply by KhorenGrig** [reply]
`import * as React from "react";
import * as ReactDOM from "react-dom";
import { useForm } from "react-hook-form";

import "./styles.css";

const defaultValues = {
  firstName: "John",
  lastName: "Doe"
};

const App = () => {
  const {
    register,
    handleSubmit,
    setValue,
    formState: { isDirty },
    reset
  } = useForm({
    defaultValues
  });

  const onSubmit = (data) => {
    alert(JSON.stringify(data));
  };

  React.useEffect(() => {
    console.log("isDirty", isDirty);
  }, [isDirty]);

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <label>First Name</label>
      <input name="firstName" type="text" ref={register} />
      <label>Last Name</label>
      <input name="lastName" type="text" ref={register({ minLength: 10 })} />
      <button
        type="button"
        onClick={() => {
          for (const [key, value] of Object.entries(defaultValues)) {
            setValue(key, value, { shouldDirty: true, shouldValidate: true });
          }
        }}
      >
        Set Values
      </button>

      <input
        style={{ display: "block", marginTop: 20 }}
        type="button"
        onClick={() => reset({ firstName: "", lastName: "" })}
        value="Reset Values"
      />
    </form>
  );
};

const rootElement = document.getElementById("root");
ReactDOM.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
  rootElement
);
`

**Reply by KhorenGrig** [reply]
This is the whole code that i have.
